src/textual/widgets/_log.py

```diff
@@ -0,0 +1,189 @@
+from __future__ import annotations
+
+import re
+from typing import TYPE_CHECKING, Iterable, Optional
+
+from rich.cells import cell_len
+from rich.highlighter import ReprHighlighter
+from rich.segment import Segment
+from rich.style import Style
+from rich.text import Text
+
+from .. import work
+from .._cache import LRUCache
+from ..geometry import Size
+from ..reactive import var
+from ..scroll_view import ScrollView
+from ..strip import Strip
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+_sub_escape = re.compile("[\u0000-\u0014]").sub
+
+
+class Log(ScrollView, can_focus=True):
+    """A widget to log text."""
+
+    DEFAULT_CSS = """
+    Log {
+        background: $surface;
+        color: $text;
+        overflow: scroll;
+    }
+    """
+
+    max_lines: var[int | None] = var[Optional[int]](None)
+    """Maximum number of lines to show"""
+
+    auto_scroll: var[bool] = var(True)
+    """Automatically scroll to new lines."""
+
+    def __init__(
+        self,
+        highlight: bool = False,
+        max_lines: int | None = None,
+        auto_scroll: bool = True,
+        name: str | None = None,
+        id: str | None = None,
+        classes: str | None = None,
+        disabled: bool = False,
+    ) -> None:
+        """Create a Log widget.
+
+        Args:
+            highlight: Enable highlighting.
+            max_lines: Maximum number of lines to display.
+            auto_scroll: Scroll to end on new lines.
+            name: The name of the text log.
+            id: The ID of the text log in the DOM.
+            classes: The CSS classes of the text log.
+            disabled: Whether the text log is disabled or not.
+        """
+        self.highlight = highlight
```

Comment:
Is the intention that this is part of the public interface? If so it should probably have a docstring.